tests/integration/test_sanity.py

```diff
@@ -522,3 +522,25 @@ def test_maxmemory(cluster):
 
     cluster.execute('config', 'set', 'maxmemory', 4000000000)
     cluster.execute('set', 'key1', val)
+
+
+def test_max_memory_eval(cluster):
+    cluster.create(3)
+
+    val = ''.join('1' for _ in range(2000000))
+
+    cluster.execute('set', 'key1', val)
+    cluster.execute('config', 'set', 'maxmemory', 100000)
+
+    with (raises(ResponseError, match="OOM command not allowed when used memory > 'maxmemory'")):
```

Comment:
reply for scripts should be `-OOM allow-oom flag is not set on the script, can not run it when used memory > 'maxmemory'`, see OSS code

Reply:
k, seems like we have two paths in redis that check OOM for scripts - can you please verify it? see https://github.com/redis/redis/blob/00097bf4aa9e28477ee5d9361569f3212fb7493a/tests/unit/moduleapi/misc.tcl#L200

Reply:
suggest talking with Oran about that, he made some changes lately in this area

Reply:
ok, can probably do that with our new FLAGS flag, as that would tell us which error message to return